### PR TITLE
Run test with QEMU same way as picolibc tests do

### PR DESCRIPTION
**Pull request by domin144**
The lit-exec-qemu.py script was modified to reproduce the QEMU calls used in picolibc's own tests.

### CODE REVIEW

File: CMakeLists.txt
Comment (mplatings) [inline CMakeLists.txt:595]:
```suggestion
        if(qemu_cpu)
```

File: CMakeLists.txt
Comment (mplatings) [inline CMakeLists.txt:598]:
```suggestion
        if(qemu_params_list)
```

File: CMakeLists.txt
Comment (mplatings) [inline CMakeLists.txt:668]:
```suggestion
    if(qemu_cpu)
```

File: CMakeLists.txt
Comment (mplatings) [inline CMakeLists.txt:671]:
```suggestion
    if(qemu_params_colon)
```

File: CMakeLists.txt
Comment (mplatings) [inline CMakeLists.txt:763]: ditto

File: CMakeLists.txt
Comment (mplatings) [inline CMakeLists.txt:766]: ditto

File: test-support/lit-exec-qemu.py
Comment (mplatings) [inline test-support/lit-exec-qemu.py:24]: Comments please explaining why each argument is necessary

File: test-support/lit-exec-qemu.py
Comment (mplatings) [inline test-support/lit-exec-qemu.py:7]: Please format this file with `black -l79` (TODO: automate this check)

**Review by mplatings — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (domin144) [reply CMakeLists.txt]: Done

File: CMakeLists.txt
Comment (domin144) [reply CMakeLists.txt]: Done

File: CMakeLists.txt
Comment (domin144) [reply CMakeLists.txt]: Done

File: CMakeLists.txt
Comment (domin144) [reply CMakeLists.txt]: Done

File: test-support/lit-exec-qemu.py
Comment (domin144) [reply test-support/lit-exec-qemu.py]: Done

File: test-support/lit-exec-qemu.py
Comment (domin144) [reply test-support/lit-exec-qemu.py:7]: Done

File: CMakeLists.txt
Comment (domin144) [reply CMakeLists.txt]: Done

File: CMakeLists.txt
Comment (domin144) [reply CMakeLists.txt]: Done

**Review by mplatings — APPROVED** [review]
LGTM but please wait until Monday morning to rebase-and-merge

**Pull request merged**
